### PR TITLE
feat: add pagination in history

### LINKED ISSUE #17: [feat] allow pagination in history to support viewing more than 10 generations
## Issue Description

Currently, we just store the generations in history. We do not limit the number of generations we store however in the `h` command we only show the latest 10 generations. Allow users to view other generations as well, this can be implemented using pagination easily.

```
/gen-gif h 0 ---> View first 10
/gen-gif h 1 ---> View next 10
/gen-gif h 2 ---> View next 10
```

### PR DESCRIPTION
**Pull request by AmanNegi**
## Description :notebook: 

- Allow users to view and store more than 10 generations.
- The user can simply pass the page number along with the command as:

 	`gen-gif h <page>`

> Fixes: #17 

## Demo :video_camera: 

https://github.com/user-attachments/assets/b5650e79-30b3-4907-a402-18113f5490bc

### CODE REVIEW

**Comment by AmanNegi** [comment]
@Nabhag8848 Could you please review the PR and let me know if any changes are required.

File: src/endpoints/GifStatusUpdateEndpoint.ts [hunk 1]
```diff
@@ -83,7 +84,10 @@ export class GifStatusUpdateEndpoint extends ApiEndpoint {
             read.getPersistenceReader()
         );
 
+        const uuid = Math.floor(Date.now() * Math.random());
```
Comment (Nabhag8848) [inline src/endpoints/GifStatusUpdateEndpoint.ts]: @AmanNegi  https://github.com/RocketChat/Apps.Notion/blob/main/src/helper/uuid.ts

File: src/persistence/GenerationPersistence.ts [hunk 2]
```diff
@@ -52,6 +53,19 @@ export class GenerationPersistence {
...
 
+    async getItemsForPage(page: number) {
+        const itemsPerPage = 10;
+        const records = await this.getAllItems();
+
+        const start = page * itemsPerPage;
+        const end = start + itemsPerPage;
+
+        const list = records.slice(start, end);
+        console.log(page, list.length, list);
```
Comment (Nabhag8848) [inline src/persistence/GenerationPersistence.ts]: @AmanNegi

**Review by Nabhag8848 — CHANGES_REQUESTED** [review]
@AmanNegi

**Comment by Nabhag8848** [comment]
@AmanNegi merge conflicts

**Comment by AmanNegi** [comment]
@Nabhag8848 Fixed all merge conflicts!

**Comment by Nabhag8848** [comment]
- @AmanNegi what happens if  **/gen-gif h** without page ? 
- what if the **/gen-gif h <random-value>** random-value page doesn't exist ? also 
- what if **/gen-gif h <some-text>** ? 

does this cases are handled ?

**Comment by AmanNegi** [comment]
If we simply use `/gen-gif h` without providing the page, by default it renders the first page.

If we provide an out of bounds page, no preview is shown suggesting that this page doesn't exist. 

(Ps: currently preview titles don't show up otherwise we could provide user with a message in preview heading itself. I have created an issue for that as well).

**Comment by Nabhag8848** [comment]
@AmanNegi 

> [!CAUTION]
> If we provide an out of bounds page, no preview is shown suggesting that this page doesn't exist.

> [!TIP]
> For now send notifications to room (which only user can see) for letting user know that.

> [!IMPORTANT]
> What if /gen-gif h "helloWorld" ? does it handles the provided page is number.

**Comment by AmanNegi** [comment]
> "For now send notifications to room (which only user can see) for letting user know that."

Okay for pages which don't have an item, we will notify the user with a message "No items to show on page 1".

> "What if /gen-gif h "helloWorld" ? does it handles the provided page is number."

Yes it does handle that case and by returns the items on page 1 itself.

File: src/handlers/PreviewerHandler.ts [hunk 3]
```diff
@@ -127,14 +127,6 @@ export class PreviewerHandler {
                     value: item.prompt,
                 };
             });
-        } else {
-            sendMessageToSelf(
-                this.modify,
```
Comment (Nabhag8848) [inline src/handlers/PreviewerHandler.ts:131]: @AmanNegi Why this has been removed ? this will break the changes we made in previous PR right ?

**Comment by Nabhag8848** [comment]
> > "For now send notifications to room (which only user can see) for letting user know that."
> 
> Okay for pages which don't have an item, we will notify the user with a message "No items to show on page 1".
> 


- For pages which don't have item - convey that there isn't any gif more than x number. so to me it makes more sense to say No page exist and to convey some message that there is only max this much page. 

> > "What if /gen-gif h "helloWorld" ? does it handles the provided page is number."
> 
> Yes it does handle that case and by returns the items on page 1 itself.

- does it gives the warning notification about that, as page is just number we have return most recent page items ?

**Comment by AmanNegi** [comment]
I have successfully fixed the above issues, and now the app shows valid error messages (for each case covered above).


#### Demo


https://github.com/user-attachments/assets/ecda7454-d213-4bd7-9d0d-80e63f877c88

**Review by Nabhag8848 — APPROVED** [review]
@AmanNegi  LGTM !

**Pull request merged**